earth_enterprise/src/fusion/autoingest/AssetVersion.h

```diff
@@ -113,6 +113,47 @@ class AssetVersionImpl : public khRefCounter, public AssetVersionStorage, public
     AssetVersionRef verref(name);
     return verref.AssetRef();
   }
+// Get the total memory used by an asset version
+  const uint64 GetSize() {
+    return (name.GetSharedStringSize()
+    + sizeof(type)
+    + sizeof(subtype)
+    + (subtype.capacity() * sizeof(char))
+    + sizeof(state)
+    + sizeof(progress)
+    + sizeof(locked)
+    + GetVectorSize(inputs)
+    + GetVectorSize(children)
+    + GetVectorSize(parents)
+    + GetVectorSize(listeners)
+    + GetVectorSize(outfiles)
+    + sizeof(meta)
+    + meta.GetSize()
+    + sizeof(beginTime)
+    + sizeof(progressTime)
+    + sizeof(endTime)
+    + sizeof(taskid)
+    + sizeof(timestamp)
+    + sizeof(filesize));
+  }
+
+// Get the total memory used by a template vector
+  template<class T>
```

Comment:
It would be nice to put these in a common location so we don't have to duplicate the code.